Libraries/acdriver/Sources/Extensions.cpp

```diff
@@ -0,0 +1,74 @@
+/**
+ Copyright (c) 2015-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#include <acdriver/Extensions.h>
+#include <libutil/Options.h>
+
+using acdriver::Extensions;
+
+static std::pair<bool, std::string>
+InsertNextImageType(Extensions::ImageTypeSet &imageTypeSet, std::vector<std::string> const &args, std::vector<std::string>::const_iterator *it)
+{
+    ext::optional<std::string> value;
+    auto ret = libutil::Options::Next<std::string>(&value, args, it);
+    if (!ret.first) {
+        return ret;
+    }
+    ext::optional<Extensions::ImageType> type = Extensions::ImageTypeFromFileExtension(*value);
+    if (!type) {
+        return std::make_pair(false, "unknown image type " + *value);
+    }
+    imageTypeSet.insert(*type);
+    return std::make_pair(true, std::string());
+}
+
+ext::optional<std::pair<bool, std::string>> Extensions::ActoolOptions::
+parseArgument(std::vector<std::string> const &args, std::vector<std::string>::const_iterator *it)
+{
+    std::string const &arg = **it;
+
+    if (arg == "--allow-extensions") {
+        return libutil::Options::Current<bool>(&_allowExtensions, arg);
+    } else if (arg == "--allow-image-types") {
+        return InsertNextImageType(_allowImageTypes, args, it);
+    } else {
+        return ext::nullopt;
+    }
+}
+
+bool Extensions::ActoolOptions::
+areOptionsValid(acdriver::Result *result) const
+{
+    if (!allowExtensions() && allowImageTypes().size() > 0) {
+        result->normal(Result::Severity::Error, "--allow-image-types requires --allow-extensions");
+        return false;
+    }
+    return true;
+}
+
+car::Rendition::Data::Format Extensions::
+ImageTypeToDataFormat(ImageType type) {
+    switch (type) {
+        case Extensions::ImageType::WEBP:
+            return car::Rendition::Data::Format::EXTENSIONS_WEBP;
+    }
+
+    abort();
+}
+
+ext::optional<Extensions::ImageType> Extensions::
+ImageTypeFromFileExtension(std::string const &extension)
+{
+    std::string lowerExtension;
+    std::transform(extension.begin(), extension.end(), std::back_inserter(lowerExtension), ::tolower);
```

Comment:
Could we use Boost's `iequals` function instead? https://stackoverflow.com/a/315463

Reply:
I don't think Boost is used in xcbuild, and I don't think this is enough for it to be added... Besides, this method way to compare is already used in xcbuild.